heron/spi/tests/java/com/twitter/heron/spi/utils/SchedulerUtilsTest.java

```diff
@@ -168,4 +175,63 @@ public void testSchedulerCommandArgs() throws Exception {
     Assert.assertArrayEquals(expectedArgs, SchedulerUtils.schedulerCommandArgs(
         Mockito.mock(Config.class), Mockito.mock(Config.class), freePorts));
   }
+
+  @Test
+  public void updatesPackingPlanWithUniformSizedContainers() {
+    SchedulerStateManagerAdaptor adaptor = Mockito.mock(SchedulerStateManagerAdaptor.class);
+    Mockito.when(adaptor.deletePackingPlan("topology")).thenReturn(true);
+    Mockito.when(adaptor.setPackingPlan(Mockito.any(PackingPlans.PackingPlan.class),
+        Mockito.eq("topology"))).thenReturn(true);
+    Config runtime = Mockito.mock(Config.class);
+    Mockito.when(runtime.get(Keys.schedulerStateManagerAdaptor())).thenReturn(adaptor);
+
+    PackingPlan packing = createPackingPlan();
+    SchedulerUtils.persistUpdatedPackingPlan("topology", packing, runtime);
+    InOrder inOrder = Mockito.inOrder(adaptor);
+    inOrder.verify(adaptor).deletePackingPlan("topology");
+    inOrder.verify(adaptor).setPackingPlan(Mockito.any(PackingPlans.PackingPlan.class),
+        Mockito.eq("topology"));
+  }
+
+  @Test
+  public void updatesPackingPlanInStateManager() {
+    PackingPlan plan = createPackingPlan();
+    PackingPlan newPlan = SchedulerUtils.getHomogenizedContainerPlan(plan);
+
+    PackingPlan.ContainerPlan[] containerArray
+        = newPlan.getContainers().toArray(new PackingPlan.ContainerPlan[2]);
+    Assert.assertEquals(2, newPlan.getContainers().size());
+
+    PackingPlan.ContainerPlan largeContainer = containerArray[0];
+    PackingPlan.ContainerPlan smallContainer = containerArray[1];
```

Comment:
The tests submits two different sized containers. Here `largeContainer` and `smallContainer` refers to the original size of the container before they are homogenized, see `SchedulerUtilsTests L#220`.